packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -118,6 +120,24 @@ library LibProposing {
             );
         }
 
+        {
+            IProofVerifier proofVerifier = IProofVerifier(
+                resolver.resolve("proof_verifier", false)
+            );
+            bool verified = proofVerifier.verifyZKP({
+                verifierId: string(
+                    abi.encodePacked(
+                        "plonk_verifier_propose",
+                        bytes32(inputs[3])
+                    )
+                ),
+                zkproof: inputs[4],
```

Comment:
```suggestion
                        bytes32(inputs[2])
                    )
                ),
                zkproof: inputs[3],
```
From the function comments above this seems to make more sense, but I may miss something.